Userland/Applications/NetworkSettings/NetworkSettingsWidget.cpp

```diff
@@ -134,33 +137,75 @@ void NetworkSettingsWidget::on_switch_enabled_or_dhcp()
 
 void NetworkSettingsWidget::apply_settings()
 {
-    auto config_file = Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes).release_value_but_fixme_should_propagate_errors();
+    auto result = apply_settings_impl();
+    if (result.is_error()) {
+        GUI::MessageBox::show_error(window(), result.release_error().string_literal());
+        return;
+    }
+}
+
+ErrorOr<void> NetworkSettingsWidget::apply_settings_impl()
+{
+    auto maybe_json = TRY(create_settings_object());
+    if (!maybe_json.has_value())
+        return {};
+    auto json = maybe_json.release_value();
+    if (json.is_empty())
+        return {};
+
+    auto pipefds = TRY(Core::System::pipe2(O_CLOEXEC));
+    ScopeGuard guard_fd1 { [&] { close(pipefds[1]); } };
+    {
+        posix_spawn_file_actions_t file_actions;
+        posix_spawn_file_actions_init(&file_actions);
+        posix_spawn_file_actions_adddup2(&file_actions, pipefds[0], STDIN_FILENO);
+
+        ScopeGuard guard_fd0_and_file_actions { [&]() {
+            posix_spawn_file_actions_destroy(&file_actions);
+            close(pipefds[0]);
+        } };
+
+        char const* argv[] = { "network-settings", nullptr };
+        (void)TRY(Core::System::posix_spawnp("network-settings"sv, &file_actions, nullptr, const_cast<char**>(argv), environ));
```

Comment:
I agree with what supercomputer7 was saying:

> The whole idea behind using the Escalator is that if you want to change settings that is only accessible from root, then you will need to pass the authentication step to do that.
Is this keeping this behavior intact?

Shouldn't this be calling into the `Escalator` app to ask permissions to run `/bin/network-settings` as root to make changes, not calling into a SUID binary? It's an explicit extra step asking "are you sure you want to change root-owned settings? here's a password prompt: ".

Reply:
If people insist, that we need a pattern to ask for the password I will create an application that does exactly that and nothing more. I can't use the Escalator here as it limits control over the target binary. I can't attach to stdin through Escalator for example.